src/main/scala/org/constellation/util/Rewards.scala

```diff
@@ -56,13 +56,15 @@ object Rewards {
       val neighborView = view.map{ case (neighbor, score) => neighborhoodReputationMatrix(key) * score }.sum
         (key, neighborView)
     }
-    weightedTransitiveReputation.mapValues{ trust => - trust * math.log(trust)/math.log(2) }
+    weightedTransitiveReputation.mapValues{ trust =>
+      if (trust == 0.0 ) 0.0
+      else - trust * math.log(trust)/math.log(2) }
```

Comment:
What I have not yet understood is what the `/log(2)` here does already. Maybe a reference would clarify the use. 

I know there's a line about logs in the economic paper, but I'm not sure if that's the one that ties to that formula. If I understand correctly, the trust does in practice vary between 0 and 1 and it's not like the trust of all nodes summed up adds to 1. I.e. the trust can indeed be not just a tiny number close to 0.

Now the function `trust * math.log(trust)/math.log(2)` starts at trust=0 and goes back to zero at trust=1. In the middle it takes its maximum at trust=1/e=0.36.. and the value there is determined by factors such as `1/log(2)`. With that choice it should be some over 0.53..

If I read it correctly, the return values of this shannonEntropy function eventually make it into the line
`val reward = partitonSize * ( 1 - trustEntropyMap(address))`
in the rewardDistribution functon. And since it's subtracted from a fixed number here (the 1), the factors such as `/log(2)` end up playing play a role. 

Does the factor f=1/log(2) have a concrete effect here, or are other factor like r*exp(1) with r between 0 and 1 thinkable.

Can you shine a light on the effects that that model ought to capture? 
E.g. with respect on how the reward for people with low trust should look like (Thinking about it since the entropy expression isn't monotonic but has a max in the middle).

I'm also interest in the model since `p * log(p)` isn't symmatric around x=1/2. (While e.g. `-((1-p) * log(1-p) - p * log(p)`) would be.